### PR TITLE
Do not leak smallrye-open-api + asm dependency management

### PR DESCRIPTION
**Pull request by snazy**
... so those do not interfere with other plugins.

### CODE REVIEW

File: settings.gradle.kts [hunk 1]
```diff
@@ -78,25 +78,20 @@ gradle.rootProject {
...
-include("jacoco")
-
-include("protobuf")
-
-include("reflection-config")
-
-include("smallrye-openapi")
-
-include("spotless")
+listOf(
```
Comment (snazy) [inline settings.gradle.kts:81]: Just style (less LOC)

File: build.gradle.kts [hunk 2]
```diff
@@ -62,151 +58,163 @@ dependencies {
...
   version = projectVersion
 
-  repositories {
-    gradlePluginPortal()
-    mavenCentral()
-    if (java.lang.Boolean.getBoolean("withMavenLocal")) {
-      mavenLocal()
+  if (project.path != ":dependency-declarations") {
+
+    repositories {
```
Comment (snazy) [inline build.gradle.kts:88]: From here up to the end of the change is just idention

File: build.gradle.kts [hunk 3]
```diff
@@ -62,151 +58,163 @@ dependencies {
...
-    api("io.smallrye:smallrye-open-api-jaxrs:$versionSmallryeOpenApi")
-    api("io.smallrye:smallrye-open-api-spring:$versionSmallryeOpenApi")
-    api("io.smallrye:smallrye-open-api-vertx:$versionSmallryeOpenApi")
     api("net.ltgt.gradle:gradle-errorprone-plugin:$versionErrorPronePlugin")
-    api("org.ow2.asm:asm:$versionAsm")
     api("org.jboss:jandex:$versionJandex")
   }
 }
 
+project(":dependency-declarations") {
```
Comment (snazy) [inline build.gradle.kts:66]: This (and the removals directly above) is the actual change.

File: build.gradle.kts [hunk 4]
```diff
@@ -38,21 +38,17 @@ val versionShadowPlugin = "7.1.2"
...
-rootProject.extra["versionJandexPlugin"] = versionJandexPlugin
-
-rootProject.extra["versionProtobufPlugin"] = versionProtobufPlugin
-
-rootProject.extra["versionQuarkus"] = versionQuarkus
-
-rootProject.extra["versionShadowPlugin"] = versionShadowPlugin
-
-rootProject.extra["versionSmallryeOpenApi"] = versionSmallryeOpenApi
+mapOf(
```
Comment (snazy) [inline build.gradle.kts:41]: Just style (less LOC)

File: settings.gradle.kts [hunk 5]
```diff
@@ -78,25 +78,20 @@ gradle.rootProject {
...
+    "checkstyle",
+    "ide-integration",
+    "publishing",
+    "jandex",
+    "jacoco",
+    "protobuf",
+    "reflection-config",
+    "smallrye-openapi",
+    "spotless",
+    "dependency-declarations"
```
Comment (XN137) [inline settings.gradle.kts:92]: wondern:
where does this come from?
i would have thought these names refer to top level folders in the repo
or are these references to projects and the last one we created on the fly?

File: build.gradle.kts [hunk 6]
```diff
@@ -38,21 +38,17 @@ val versionShadowPlugin = "7.1.2"
...
+    "versionAsm" to versionAsm,
+    "versionErrorPronePlugin" to versionErrorPronePlugin,
+    "versionJandex" to versionJandex,
+    "versionJandexPlugin" to versionJandexPlugin,
+    "versionProtobufPlugin" to versionProtobufPlugin,
+    "versionQuarkus" to versionQuarkus,
+    "versionShadowPlugin" to versionShadowPlugin,
+    "versionSmallryeOpenApi" to versionSmallryeOpenApi
```
Comment (XN137) [inline build.gradle.kts:49]: nit: trailing comma in these collection literals maybe

**Review by XN137 — APPROVED** [review]

File: build.gradle.kts (hunk 6 above)
Comment (snazy) [reply build.gradle.kts:49]: that's a syntax error

File: settings.gradle.kts (hunk 5 above)
Comment (snazy) [reply settings.gradle.kts:92]: Those are just project names - not related to the project directories. See Spark-extensions projects in Nessie - same project directory shared for multiple projects.

**Pull request merged**